### PR TITLE
Add support for importing self-signed certificate to ACM

### PR DESCRIPTION
**Pull request by r351574nc3**
## what
* ACM support allows using `aws_acm_certificate` TF resource to store certificates and keys directly in ACM for use with other AWS services that access certificates in that way.

## why
* Support the following use case: VPN client uses self-signed-cert to create VPN endpoints using certificate based authentication and storing certificates in ACM

## references
* N/A

### CODE REVIEW

File: examples/acm/versions.tf [hunk 1]
```diff
@@ -0,0 +1,14 @@
+terraform {
+  required_version = ">= 0.13.0"
+
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = ">= 2.0"
```
Comment (nitrocode) [inline examples/acm/versions.tf:7]:
```suggestion
      version = ">= 3.0"
```

File: acm.tf [hunk 2]
```diff
@@ -0,0 +1,6 @@
+resource "aws_acm_certificate" "root" {
```
Comment (osterman) [inline acm.tf:1]:
```suggestion
resource "aws_acm_certificate" "default" {
```

**Review by korenyoni — COMMENTED** [review]
A few comments:

1) Please append to README.yaml some information on the ACM output.
2) I don't think ACM is a secrets store, so it shouldn't be an applicable value in `secrets_store_type`. Although this control switch is quite elegant,  so understand why you did it this way. I think it should write to SSM or ASM regardless, with a separate `import_to_acm_enabled` control switch.
3) You are missing an output with the ACM certificate's ARN.
4) If you are up for the task, you could modify Terratest to read the ARN output (see above), then read from ACM and confirm that the body of the cert matches.

Only 1-3 are required from my point of view.

**Review by korenyoni — DISMISSED** [review]
See https://github.com/cloudposse/terraform-aws-ssm-tls-self-signed-cert/pull/5#pullrequestreview-750675345

**Comment by osterman** [comment]
That was my call to use that flag. It's the storage back end. We can store in SSM, acm, acm, s3 etc. what's the problem? :-)

**Comment by korenyoni** [comment]
@osterman I also don't believe you are able to retrieve the private key for a cert from ACM [unless it was issued by ACM Private CA](https://docs.aws.amazon.com/acm/latest/userguide/export-private.html). Because of this, I do not consider it a secrets store, because it cannot store the private key secret.

**Comment by r351574nc3** [comment]
> @osterman I also don't believe you are able to retrieve the private key for a cert from ACM unless it was issued by ACM Private CA. Because of this, I do not consider it a secrets store, because it cannot store the private key secret.

Even if we can't retrieve it, it's available to other AWS services. ACM is used exclusively for certificates by some AWS services. The case that I need it for is VPN endpoints.

**Comment by korenyoni** [comment]
> Even if we can't retrieve it, it's available to other AWS services. ACM is used exclusively for certificates by some AWS services. The case that I need it for is VPN endpoints.

@r351574nc3 makes sense. In that case you can dismiss my suggestion to make a separate control switch for ACM. But I suggest to make a `certificate_backends` list variable or something similar, because of the following reasons:

1) The only two true secrets stores in AWS (in my opinion, since it's a loose definition) are AWS SSM Parameter Store or AWS Secrets Manager, because both of these support arbitrary secrets.
2) If someone needs to export to ACM and also needs the private key handy, it should be possible to deploy to a combination of the tree backends, instead of only one (since you cannot get the private key from ACM).

**Comment by osterman** [comment]
> var.certificate_backends
+1

**Comment by r351574nc3** [comment]
@korenyoni Could I get a final review on this? Terratests are passing and requested changes were made.

**Review by osterman — APPROVED** [review]

File: examples/preexisting_key/variables.tf [hunk 3]
```diff
@@ -13,7 +13,7 @@ variable "private_key_algorithm" {
   type        = string
 }
 
-variable "secrets_store_base64_enabled" {
+variable "certificate_backends_base64_enabled" {
   description = "Enable or disable base64 encoding of secrets before writing them to the secrets store."
```
Comment (korenyoni) [inline examples/preexisting_key/variables.tf:17]: We should document that base64 encoding does not apply to the ACM backend (in another PR)

**Pull request merged**